tests/integration/test_charm.py

```diff
@@ -142,8 +140,86 @@ async def test_get_primary_action(ops_test: OpsTest) -> None:
         assert identified_primary == expected_primary
 
 
-async def test_cluster_is_stable_after_deletion(ops_test: OpsTest) -> None:
-    """Tests that the cluster behavior after planned non-leader unit removal."""
+async def test_cluster_is_stable_after_leader_deletion(ops_test: OpsTest) -> None:
+    """Tests that the cluster cluster behavior after planned leader unit removal.
+
+    This test verifies that the behavior of when a leader is deleted that the new leader, on
+    calling leader_elected will reconfigure the replicaset. Similarly, this tests the case of a
+    primary steping down, since on deployment it is maintained that the leader is primary.
+    """
+    # find & destroy leader unit
+    # grab leader unit
+    leader_unit = await find_leader_unit(ops_test)
+
+    # verify that we have a leader
+    assert leader_unit is not None, "No unit is leader"
+
+    # save ip and delete leader
+    leader_ip = leader_unit.public_address
+    await leader_unit.destroy()
+
+    # wait for app to be active after removal of leader
+    # issuing dummy update_status to re-trigger event
+    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
+    # reset update_status
+    await ops_test.model.set_config({"update-status-hook-interval": "5m"})
+
+    # verify that there are two units running after deletion of leader
+    assert len(ops_test.model.applications[APP_NAME].units) == 2
+
+    # verify that a new leader has been elected
+    # grab leader unit
+    leader_unit = await find_leader_unit(ops_test)
+
+    # verify that we have a leader
+    assert leader_unit is not None, "No unit is leader"
+
+    # grab remaining IPS, maintaining that the deleted ip is not in them
+    ip_addresses = []
+    for unit in ops_test.model.applications[APP_NAME].units:
+        if not unit.public_address == leader_ip:
```

Comment:
Is this really needed? I ask because the old leader unit was already removed, so it's not in the units list and there is an equivalent check on L210.

Reply:
The other check is checking the mongo conf which is different. But I agree this doesn't need to be checked, :)